### ISSUE TITLE
private standard Unity MonoBehaviour methods generate IDE0051

### ISSUE THREAD

**Issue by WarpedPixel** [microsoft/Microsoft.Unity.Analyzers#170]
(Hopefully this is the right way to report what looks like a bug to me. Love the analyzers and trying to get to zero warning in my code, thanks!)
#### Bug description
This code generates IDE0051 with analyzer 1.10.2:

```csharp
public class UserInput : MonoBehaviour
{
  void Start() // this line triggers IDE0051
  {
        InitializeMouse();
  }
  public void Update() // this line does NOT trigger IDE0051
  {
        GetMouseClick();
  }
}
```

#### Expected behavior
These methods are not "unused" since, when derived from MonoBehaviour, they are actually invoked by Unity. 
This is regardless if they are public or private.
So the expected behavior is for all these methods with standard Unity MonoBehavior signatures (Start, Awake, Update, etc.) to never flag IDE0051. While other private methods continue to flag that normally.

#### Workaround
Just make sure all your standard MonoBehaviour methods are declared public

**Comment by WarpedPixel** [comment]
You have indeed. Running vscode latest on Mac Catalina latest. Unity 2019.4.23f1. I had to patch my csproj files to limit LangVersion to 8.0 since that is what Unity supports. I definitely see other UNTxxxx warnings. For example, I see UNT0001 if the Update body in my example above is empty. I'll test a few things.

**Comment by sailro** [comment]
Ah that's why: vscode is not yet able to use suppressors (USPxxxx), only regular analyzers (UNTxxxx).

The original issue is tracked here:
https://github.com/OmniSharp/omnisharp-roslyn/issues/1711

**Comment by WeirdBeardDev** [comment]
@WarpedPixel [here](https://code.visualstudio.com/docs/other/unity#_enabling-unity-warnings) are directions on how to setup an .editorconfig file and then turn off IDE0051.

@sailro sweet, I've been waiting for this to be addressed.  Thanks!

**Comment by enkeyz** [comment]
I have the same issue, looks like something broke again. VStudio 2022 marks Monobehaviour methods - Start, Update, etc - as unused methods.
